src/com/homepractice/AbstractionPractice.java

```diff
@@ -0,0 +1,35 @@
+package com.homepractice;
+
+// You can not create object of abstract class..so what is the need of abstract class constructor
+// if abstract class contains 50 fields and also has 100 child class then without abstract class constructor 
+//you have to write initialization in every child class ..reduces the code reusablity
+public abstract class AbstractionPractice {
+
+	 String name;
+	 int age;
+	
+	public AbstractionPractice(String name, int age) {
+		super();
```

Comment:
What all is there in object constructor? and what methods are there?

Reply:
1 object clone();-- to create copy of this object
2. boolean equals(object o)-- to compare two objects
3. void finalize()--called by garbage collector on an object when GC determines there is no more ref to the obj.
4. class <>getClass()--returns run time class of an object
5. int hashCode()--returns hashcode value of an obj
6. void notify()--Wakes up a single thread that is waiting on this object's monitor.
7.void  notifyAll()--
8.  String toString()--Returns a string representation of the object.
9. void wait()---Causes the current thread to wait until another thread invokes the notify() method